auth/src/main/java/feast/auth/config/CacheConfiguration.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.auth.config;
+
+import com.google.common.cache.CacheBuilder;
+import feast.auth.utils.AuthUtils;
+import java.lang.reflect.Method;
+import java.util.concurrent.TimeUnit;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CachingConfigurer;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.concurrent.ConcurrentMapCache;
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager;
+import org.springframework.cache.interceptor.CacheErrorHandler;
+import org.springframework.cache.interceptor.CacheResolver;
+import org.springframework.cache.interceptor.KeyGenerator;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.Authentication;
+
+/** CacheConfiguration class defines Cache settings for HttpAuthorizationProvider class. */
+@Configuration
+@EnableCaching
+@Setter
+@Getter
+public class CacheConfiguration implements CachingConfigurer {
+
+  private static final int CACHE_SIZE = 10000;
+
+  public static int TTL = 60;
+
+  public static final String AUTHORIZATION_CACHE = "authorization";
+
+  @Autowired SecurityProperties secutiryProps;
+
+  @Bean
+  public CacheManager cacheManager() {
+    ConcurrentMapCacheManager cacheManager =
+        new ConcurrentMapCacheManager(AUTHORIZATION_CACHE) {
+
+          @Override
+          protected Cache createConcurrentMapCache(final String name) {
+            return new ConcurrentMapCache(
+                name,
+                CacheBuilder.newBuilder()
+                    .expireAfterWrite(TTL, TimeUnit.SECONDS)
+                    .maximumSize(CACHE_SIZE)
+                    .build()
+                    .asMap(),
+                false);
+          }
+        };
+
+    return cacheManager;
+  }
+
+  @Override
+  public CacheResolver cacheResolver() {
+    // TODO Auto-generated method stub
+    return null;
+  }
+
+  /*
+   * KeyGenerator for HttpAuthorizationProvider.checkAccessToProject() Method.
```

Comment:
Can you update this comment to be a bit more descriptive?

Reply:
@cacheable, By default uses concatenated method params as key. For this case since it made sense to cache based on specific field in `Authentication` object(subClaim). Also we could set `KeyGenerator` in  `@cacheable`. I could have made this a bean and mentioned it in `@cacheable`. I choose to make it default, since i didn't see any other caches that we would add to auth module.

Reply:
Yes, but my point is that `CacheConfiguration` is not `ProjectAuhorizationCacheConfiguration`. It's a general purpose class. This method is called `keyGenerator` which is also generic. However, the implementation is specific to a single caller (project authorization).

I understand your reasoning, but I would prefer that we either create a specific bean and select `keyGenerator` in `@cacheable` or have conditional logic here in the `KeyGenerator` with specific key generation methods.